src/components/Layouts/_index.scss

```diff
@@ -0,0 +1,104 @@
+////
+/// Modular layouts styles.
+/// @group modular-layouts
+/// @copyright IBM Security 2020
+////
+
+@import '@carbon/colors/scss/colors';
+@import '@carbon/layout/scss/spacing';
+@import '@carbon/themes/scss/theme-maps';
+
+/// Generate a map of spacing tokens from the provided Carbon theme map.
+/// @access private
+/// @param {Map} $theme-map - Map of theme variables.
+/// @returns {Map} - The filtered map of only spacing variables.
+@function get-spacing-tokens-map($theme-map) {
+  $-map: ();
+
+  @each $token, $value in $theme-map {
+    @if type-of($value) == 'number' {
+      $-map: map-merge(
+        $-map,
+        (
+          $token: $value,
+        )
+      );
+    }
+  }
+
+  @return $-map;
+}
+
+/// Map of spacing tokens.
+/// @type Map<String, Number>
+/// @access private
+$spacing-tokens-map: get-spacing-tokens-map($carbon--theme--g100);
+
+// Generate styles for the debugging message.
+@mixin debug {
+  @include carbon--type-style($name: caption-01);
+  position: absolute;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  padding: $spacing-02;
+  left: 0;
+  color: $carbon--black-100;
+  background-color: $carbon--magenta-40;
+}
+
+// Generate "top" padding classes and styles for each spacing token.
+@mixin spacing-top {
+  @each $token, $value in $spacing-tokens-map {
+    .spacing--#{$token}--top {
+      padding-top: $value;
+      position: relative;
+    }
+  }
+}
+
+// Generate debugging styles for "top" padding classes.
+@mixin debug-spacing-top {
+  @each $token, $value in $spacing-tokens-map {
+    .spacing--#{$token}--top::before {
+      @include debug;
+      content: 'spacing--#{$token}--top';
+      height: $value;
+      top: 0;
+    }
+  }
+}
+
+// Generate "bottom" padding classes and styles for each spacing token.
+@mixin spacing-bottom {
+  @each $token, $value in $spacing-tokens-map {
+    .spacing--#{$token}--bottom {
+      padding-bottom: $value;
+      position: relative;
+    }
+  }
+}
+
+// Generate debugging styles for "bottom" padding classes.
+@mixin debug-spacing-bottom {
+  @each $token, $value in $spacing-tokens-map {
+    .spacing--#{$token}--bottom::after {
+      @include debug;
+      content: 'spacing--#{$token}--bottom';
+      height: $value;
+      bottom: 0;
+    }
+  }
+}
```

Comment:
Given there are multiple `@each` directives here on the same map with similar properties, it feels like an opportunity for a mild refactor.

Reply:
I'm also curious whether you foresee these mixins being leveraged for other use cases?

Reply:
Yes these can likely be combined. 👍

But no I don’t see how these could be used outside of this context.